### PR TITLE
Merge mail-deploy branch with master

### PR DESCRIPTION
**Pull request by dibyadas**
The mail-deploy branch has the features to generate the profile pages of the members as well as the feature to send mail to us when someone fills up the Contact Us page.

### CODE REVIEW

**Comment by dibyadas** [comment]
@pranitbauva1997 You can ignore the templates/t.html and templates/t2.html  while reviewing. They are present for debugging purposes.

File: app.py [hunk 1]
```diff
@@ -11,60 +12,74 @@
...
 
-@app.route('/mail', methods=['POST'])
-def mail():
+def send_mail(form_msg,form_name,form_email):
+	print("inside async")
 	app.config['MAIL_SERVER']='smtp.gmail.com'
 	app.config['MAIL_PORT'] = 587      #465 for SSL
 	app.config['MAIL_USE_TLS'] = True
 	app.config['MAIL_USE_SSL']=  False #True
 	app.config['MAIL_USERNAME'] = 'dibyadasiscool@gmail.com'
```
Comment (dibyadas) [inline app.py]: This is a test email id. It will changed once the development is done.

File: app.py [hunk 2]
```diff
@@ -11,60 +12,74 @@
...
 	app.config['MAIL_PASSWORD'] = 'samsunghp'
+	print("all config done")
 	mail = Mail(app)
-	form_data = request.form['msg']
+	print("init mail app")
+	form_data = form_msg          #request.form['msg']
+	print("message taken")
 	msg = Message('Query',sender = 'dibyadasiscool@gmail.com', recipients = ['dibyadas998@gmail.com'])
```
Comment (dibyadas) [inline app.py]: Proper email ID will be replaced, once done with development.

**Comment by pranitbauva1997** [comment]
I am not the perfect person to review flask code.

**Comment by dibyadas** [comment]
@pranitbauva1997 No worries. Thanks for you time. :)

File: app.py [hunk 3]
```diff
@@ -1,43 +1,95 @@
...
+    # list to be passed while rendering & some more credentials can be added
     repogen(name)
-    return render_template('test.html', name=name)
+    repo_dict = dict(zip(repos, repolink))
+    owner = requests.get("https://api.github.com/users/" + name)
+    owner_data = json.loads(owner.content.decode("utf-8"))
+    x = render_template('temp.html', name=name, image=image,
+                        repo_dict=repo_dict, owner_data=owner_data)
+    with open("templates/t2.html", "w") as f:
+        f.write(x)
```
Comment (DefCon-007) [inline app.py:50]: @dibyadas Why are we saving the data to a file here ?

File: app.py [hunk 4]
```diff
@@ -1,43 +1,95 @@
...
+
+
+def send_mail(form_msg, form_name, form_email):
+    print("inside async")
+    app.config['MAIL_SERVER'] = 'smtp.gmail.com'
+    app.config['MAIL_PORT'] = 587  # 465 for SSL
+    app.config['MAIL_USE_TLS'] = True
+    app.config['MAIL_USE_SSL'] = False  # True
+    app.config['MAIL_USERNAME'] = 'dibyadasiscool@gmail.com'
+    app.config['MAIL_PASSWORD'] = 'samsunghp'
```
Comment (DefCon-007) [inline app.py:62]: I seriously hope this is not your real password !

Comment (dibyadas) [reply app.py:62]: Haha. This is a test email ID  I am using for development purposes. Once done, I will change that.

File: app.py (hunk 3 above)
Comment (dibyadas) [reply app.py:50]: For testing. Once done with development , will delete this code.

**Comment by dibyadas** [comment]
Closed via commit #[f0c8386](https://github.com/kossiitkgp/website-mail/commit/f0c838691994fe4fad288509f96ae59d367c5a6e)
As per @DefCon-007 's instructions, the mail sending part is now separated from the profile generator part.

**Pull request closed without merging**